src/permutat.cc

```diff
@@ -418,9 +431,18 @@ static Obj LQuoPerm(Obj opL, Obj opR)
     const TR *          ptR;            /* pointer to the right operand    */
     UInt                p;              /* loop variable                   */
 
-    /* compute the size of the result and allocate a bag                   */
     degL = DEG_PERM<TL>(opL);
     degR = DEG_PERM<TR>(opR);
+
+    // handle trivial cases
+    if (degL == 0) {
+        return opR;
+    }
+    if (degR == 0) {
+        return INV(opL);
```

Comment:
We could slightly optimize this to:
```suggestion
        return InvPerm<T>(opL);
```
I think it will also need a declaration for `InvPerm` at the top.